src/client/dfuse/il/int_posix.c

```diff
@@ -168,90 +180,161 @@ ioil_init(void)
 	struct rlimit rlimit;
 	int rc;
 
+	rc = daos_init();
+	if (rc)
+		return;
+
 	pthread_once(&init_links_flag, init_links);
 
 	/* Get maximum number of file descriptors */
 	rc = getrlimit(RLIMIT_NOFILE, &rlimit);
 	if (rc != 0) {
 		DFUSE_LOG_ERROR("Could not get process file descriptor limit"
 				", disabling kernel bypass");
+		printf("Failed\n");
 		return;
 	}
 
 	rc = ioil_initialize_fd_table(rlimit.rlim_max);
 	if (rc != 0) {
 		DFUSE_LOG_ERROR("Could not create fd_table, rc = %d,"
 				", disabling kernel bypass", rc);
+		printf("Failed.\n");
 		return;
 	}
 
-	DFUSE_LOG_INFO("Using IONSS: dfuse_prefix at %s, dfuse_id is %d",
-		       dfuse_prefix, dfuse_id);
-
-	__sync_synchronize();
+	rc = pthread_mutex_init(&ioil_ioc.ioc_lock, NULL);
+	if (rc)
+		return;
 
-	ioil_initialized = true;
+	ioil_ioc.ioc_initialized = true;
 }
 
 static __attribute__((destructor)) void
 ioil_fini(void)
 {
-	if (ioil_initialized) {
-		free(projections);
-	}
-	ioil_initialized = false;
+	ioil_ioc.ioc_initialized = false;
 
-	__sync_synchronize();
+	if (ioil_ioc.ioc_open_fd_count > 0) {
```

Comment:
i understand from this that we only support single pool and container currently?